src/components/TextInput/TextInput.js

```diff
@@ -523,6 +526,9 @@ class TextInput extends React.Component<Props, State> {
     const { value = '', selectorValue = {}, label } = inputProps;
     const { selector = {}, input: inputValue } = selectorValue;
     const textInputValue = inputValue || value;
+    const { genericToken: _genericToken } = images(theme);
+    const genericToken = fallbackToGenericToken && _genericToken;
+    const fallbackSource = genericToken || _fallbackSource;
```

Comment:
Naming looks very complex, maybe this would make it read simpler (plus suggesested change above)?
```suggestion
    if (fallbackToGenericToken) {
        const { genericToken: defaultGenericToken } = images(theme);
        fallbackSource = defaultGenericToken;
    }
```